frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3615,6 +3615,11 @@ bool TypeInference::onlyBitsOrBitStructs(const IR::Type *type) const {
     return false;
 }
 
+const IR::Node *TypeInference::postorder(IR::MethodCallStatement *mcs) {
+    // Remove mcs if child methodCall resolves to a compile-time constant.
```

Comment:
AFAIK the only way for `methodCall` to be `nullptr` is if it resolved to a compile-time constant, which means it cannot possibly have side-effects.

Reply:
Okay, I would rephrase that the methodcall can only be nullpointer if it is resolved to a compile-time constant.

Reply:
I suppose that one could invent an extern function that returns some compile-time constant AND produces some side-effect(s), but no such function exists. If it did exist, then it would be incorrect for the existing transformations to simply replace its usage with a compile-time constant in the first place, so this PR's changes would not be relevant.

Reply:
Yeah, I was thinking of example like this but then realized that this constant inlining is only possible for a very specific set of functions. General externs with inout arguments or functions are not inlined to a constant iirc. Or not in the type checker at least.